### ISSUE TITLE
Is it safer than crypto-js?

### ISSUE THREAD

**Issue by TChengZ** [originjs/crypto-js-wasm#10]
i have seen the code in `algo.aes.test.js`
`
const key128 = cryptoJs.enc.Hex.parse('000102030405060708090a0b0c0d0e0f');
const key192 = cryptoJs.enc.Hex.parse('000102030405060708090a0b0c0d0e0f1011121314151617');
const key256 = cryptoJs.enc.Hex.parse('000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f');
const iv = cryptoJs.enc.Hex.parse('101112131415161718191a1b1c1d1e1f');
`
first, since the key and iv are written in js code, everyone can watch it by console in browser just like use crypto-js。
second, the encryption algorithms are all same with crypto-js like aes,rsa and so on。
so i real doubt weather crypto-js-wasm is safer than crypto-js。

maybe the key and iv should put in wasm is safer...

**Comment by PeterAlfredLee** [comment]
Hi @TChengZ 

We want to be as much as possible to be compatible with `crypto-js`, which means we need to provide the same interfaces as `crypto-js` did.
The key and iv can not be hidden as javascript do not have any mechanism that can help with this. We can only hide the procedure of encryption as it is run in wasm.

Besides this, when reacting with javascript code, the wasm code need the help of some so called `glue code` - which is also javascript code. Which means you can also obtain the key and iv from the `glue code`, cause it's also written in javascript.

In other words, when reacting with javascript, `wasm` can not hide everything. Maybe pure wasm without any javascript `glue code` can be supported by browsers in the future, but it's another story but not `crypto-js-wasm`.

If you got any better ideas, feel free to let me know. And a PR is always welcome. :)
